### PR TITLE
#23 Use Retrofit instead of Volley to make Network requests

### LINKED ISSUE #23: Use Retrofit instead of Volley to make Network requests
Volley isn't an industry standard now, So i think we should move to using Retrofit to make Network requests.

### PR DESCRIPTION
**Pull request by eXeeLD**
resolves #23

### CODE REVIEW

File: app/src/main/java/com/example/srv_twry/studentcompanion/Fragments/CodingCalendarListFragment.java [hunk 1]
```diff
@@ -96,7 +96,7 @@ private void startLoadingData() {
 
         //If the device is online then get the updated data from the server otherwise use the cached data from the database.
         if (isOnline()){
-            FetchContestsVolley fetchContestsVolley = new FetchContestsVolley(getContext(),this);
+            FetchContestsRetrofit fetchContestsVolley = new FetchContestsRetrofit(getContext(),this);
```
Comment (dr0pdb) [inline app/src/main/java/com/example/srv_twry/studentcompanion/Fragments/CodingCalendarListFragment.java]: Rename fetchContestsVolley to fetchContestsRetrofit.

File: app/src/main/java/com/example/srv_twry/studentcompanion/Network/FetchContestsRetrofit.java [hunk 2]
```diff
@@ -3,57 +3,68 @@
...
-    public  void fetchContest(){
-        JsonObjectRequest jsonObjectRequest = new JsonObjectRequest(Request.Method.GET, HACKERRANK_API_URL, null, new Response.Listener<JSONObject>() {
+    public void fetchContest(){
+
+        api.getFeed().enqueue(new Callback<ModelsPOJO>() {
             @Override
-            public void onResponse(JSONObject response) {
-                addToDatabase(response);
+            public void onResponse(@NonNull Call<ModelsPOJO> call, @NonNull Response<ModelsPOJO> response) {
+
```
Comment (dr0pdb) [inline app/src/main/java/com/example/srv_twry/studentcompanion/Network/FetchContestsRetrofit.java]: Don't add unnecessary lines.

File: app/src/main/java/com/example/srv_twry/studentcompanion/Network/FetchContestsRetrofit.java [hunk 3]
```diff
@@ -3,57 +3,68 @@
...
+                addToDatabase(response.body().getModels());
                 onLoadingFinishedListener.onLoadingFinished();
             }
-        }, new Response.ErrorListener() {
+
             @Override
-            public void onErrorResponse(VolleyError error) {
+            public void onFailure(@NonNull Call<ModelsPOJO> call, @NonNull Throwable t) {
+
```
Comment (dr0pdb) [inline app/src/main/java/com/example/srv_twry/studentcompanion/Network/FetchContestsRetrofit.java]: Remove this extra line.

File: app/src/main/java/com/example/srv_twry/studentcompanion/Network/FetchContestsRetrofit.java [hunk 4]
```diff
@@ -70,29 +81,25 @@ private void addToDatabase(JSONObject response) {
...
-            JSONArray models = response.getJSONArray("models");
             //The hackerrank API sorts contests in reverse order so reversing the array
-            for(int i=models.length()-1; i>=0; i--){
-                JSONObject obj = models.getJSONObject(i);
-                String title = obj.getString("title");
-                String description = obj.getString("description");
-                String start = obj.getString("start");
-                String end = obj.getString("end");
-                String url = obj.getString("url");
+            for(int i=response.size()-1; i>=0; i--){
```
Comment (dr0pdb) [inline app/src/main/java/com/example/srv_twry/studentcompanion/Network/FetchContestsRetrofit.java]: Add proper spaces please. 
It should be 
``` Java
for (int i = response.size() - 1; i >= 0; i--) {
    //code
}
```

File: app/src/main/java/com/example/srv_twry/studentcompanion/Network/FetchContestsRetrofit.java [hunk 5]
```diff
@@ -70,29 +81,25 @@ private void addToDatabase(JSONObject response) {
...
 
                 ContentValues contentValues = new ContentValues();
-                contentValues.put(DatabaseContract.ContestEntry.CONTEST_COLUMN_TITLE,title);
-                contentValues.put(DatabaseContract.ContestEntry.CONTEST_COLUMN_DESCRIPTION,description);
-                contentValues.put(DatabaseContract.ContestEntry.CONTEST_COLUMN_URL,url);
-                contentValues.put(DatabaseContract.ContestEntry.CONTEST_COLUMN_START_TIME,start);
-                contentValues.put(DatabaseContract.ContestEntry.CONTEST_COLUMN_END_TIME,end);
+                contentValues.put(DatabaseContract.ContestEntry.CONTEST_COLUMN_TITLE, response.get(i).getTitle());
```
Comment (dr0pdb) [inline app/src/main/java/com/example/srv_twry/studentcompanion/Network/FetchContestsRetrofit.java:86]: Add space after comma in each of these lines.

File: app/src/main/java/com/example/srv_twry/studentcompanion/Network/NetworkPOJOs/FeedPOJO.java [hunk 6]
```diff
@@ -0,0 +1,30 @@
+package com.example.srv_twry.studentcompanion.Network.NetworkPOJOs;
+
+public class FeedPOJO {
+
```
Comment (dr0pdb) [inline app/src/main/java/com/example/srv_twry/studentcompanion/Network/NetworkPOJOs/FeedPOJO.java]: No need for this line.

**Review by dr0pdb — CHANGES_REQUESTED** [review]
There are few changes that need to be made. They are all formatting ones. 
One thing i would like to say is that the formatting in this project is not very good, since i did this project some time ago now and i have learnt a lot since by contributing to other open source projects. 
Hence from now, we should try to use proper formatting.

File: app/src/main/java/com/example/srv_twry/studentcompanion/Network/FetchContestsRetrofit.java (hunk 2 above)
Comment (eXeeLD) [reply app/src/main/java/com/example/srv_twry/studentcompanion/Network/FetchContestsRetrofit.java]: i will correct the changes , but , doesn't this add readability to the code base ?

**Comment by eXeeLD** [comment]
resolves #23

**Review by dr0pdb — APPROVED** [review]

**Pull request merged**